src/modules/DNS.rb

```diff
@@ -668,6 +668,16 @@ def create_hostname_link
       nil
     end
 
+    # Checks if system DHCLIENT_SET_HOSTNAME is valid
+    #
+    # @return [Boolean]
+    def valid_dhcp_cfg?
+      return false if LanItems.find_set_hostname_ifaces.size > 1
```

Comment:
This new method is the only one that refers to LanItems (without even importing it). Maybe it should be moved there.